src/localstream.js

```diff
@@ -489,6 +557,33 @@ const localstream = (function() {
 		 */
 		close: () => {
 			close();
+		},
+		/**
+		 * Switches between cameras.
+		 */
+		switchCamera: () => {
+			_switchCamera();
+		},
+		/**
+		 * Adds the audio-and-video listener which will be called after the audio-and-video stream is established used by switching camera feature
+		 * @param cb - The audio and video listener to add 
+		 */
+		addStreamAudioVideoListener: (cb) => {
+			streamListenersAudioVideo.push(cb);
+		},
+        /**
+         * getting all of the video sources/cameras that are connected to the current device/browser
+         */
+		getAllVideoSources:() =>{
+			_getAllVideoSources();
+			return videoSources;
+		},
+        /**
+         * getting the number of all of the video sources/cameras that are connected to the current device/browser
+         */
+		getVideoSourceNumber:() =>{
```

Comment:
`_getAllVideoSources` is asynchronous so `videoSources` can be empty if `enumerateDevices` takes time. Use a `Promise` base approach